var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -127,7 +134,9 @@ def install(self, spec, prefix):
                 'complex' if '+complex' in spec else 'real'),
             '--with-shared-libraries=%s' % ('1' if '+shared' in spec else '0'),
             '--with-debugging=%s' % ('1' if '+debug' in spec else '0'),
-            '--with-blas-lapack-dir=%s' % spec['lapack'].prefix
+            '--with-lapack-lib=%s' % spec['lapack'].lapack_static_lib,
```

Comment:
@alalazo sure, you can break the build in run-time via wrong flags. My points was -- it is **a** solution. Whereas yours is another one.

@citibeth for the sake of this PR i would stick the shared (if defined) and static otherwise. Switching Blas/Lapack to lists will not happen immediately and requires a lot of testing.